### ISSUE TITLE
Bad Error Handling if Backend is unavailable

### ISSUE THREAD

**Issue by vlerenc** [gardener/dashboard#77]
I was in the Garden project (down below you see the seeds), then switched to another (datahub, which is not running our seeds ;-). I saw an error message and still saw the old clusters from the Garden project:
<img width="1279" alt="screen shot 2018-05-03 at 10 11 45" src="https://user-images.githubusercontent.com/4974203/39566018-6d2890b4-4eba-11e8-9701-2d8ca8bbfe75.png">
Neither is good, the error in the first place and the clusters from the wrong project thereafter.

**Comment by vlerenc** [comment]
Ah, when reloading the Dashboard, I get 504 Gateway Time-out. It seems we have an issue that was triggering the first error above, so what remains is the follow-up error (title of issue focus only on that).

**Comment by grolu** [comment]
I added error handling for the Login / Start page (see #81). Moreover, I added error handling in case backend requests fail, to make sure that the local store gets cleared and we do not show data belonging to the old namespace / project after switching to another.
We still have very poor error handling from a users point of view, we basically only display the error code / name via a toast / snackbar, e.g.
`Request failed with status code 504`
However, this is probably sufficient for now. All things mentioned in this and the referenced issue have been fixed.
